src/m365/purview/commands/retentioneventtype/retentioneventtype-remove.spec.ts

```diff
@@ -0,0 +1,168 @@
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    promptOptions = undefined;
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+    atStub = sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(false);
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      accessToken.isAppOnlyAccessToken,
+      Cli.prompt,
+      request.delete
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.RETENTIONEVENTTYPE_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if id is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('validates for a correct input with id', async () => {
+    const actual = await command.validate({ options: { id: validId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified retention event type when confirm option not passed', async () => {
+    await command.action(logger, { options: { id: validId } });
+
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified retention event type when confirm option not passed and prompt not confirmed', async () => {
+    const deleteSpy = sinon.spy(request, 'delete');
+    await command.action(logger, { options: { id: validId } });
+    assert(deleteSpy.notCalled);
+  });
+
+  it('correctly deletes retention event type by id', async () => {
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/security/triggerTypes/retentionEventTypes/${validId}`) {
+        return;
+      }
+
+      throw 'Invalid Request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    await command.action(logger, { options: { id: validId } });
+  });
+
+  it('correctly deletes retention event type by id when prompt confirmed', async () => {
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/security/triggerTypes/retentionEventTypes/${validId}`) {
+        return;
+      }
+
+      throw 'Invalid Request';
+    });
+
+    await command.action(logger, { options: { id: validId, confirm: true } });
+  });
+
+  it('throws an error when we execute the command using application permissions', async () => {
+    atStub.restore();
```

Comment:
Instead of this, I believe we can simply use `sinonUtil.restore(accessToken.isAppOnlyAccessToken)`.

You won't need the atStub property.

Reply:
I consulted my friend @milanholemans and he told me to do it this way 😄 I can revert it though.

Reply:
Potayto, potahto 😉

Reply:
It's a minor point of course, but this way makes it slightly more difficult to understand what's going on. Also we need an unnecessary extra line of code. (The `let atStub`) Thirdly we don't do this in a lot of places. Let's just be consistent with the main route of how we work.

If it's up to me I'd say: change it to `sinonUtil.restore(accessToken.isAppOnlyAccessToken)`